crates/pallets/starknet/src/tests/utils.rs

```diff
@@ -112,3 +116,58 @@ pub fn build_get_balance_contract_call(account_address: StarkFelt) -> (EntryPoin
 
     (balance_of_selector, calldata)
 }
+
+pub fn create_resource_bounds() -> starknet_api::transaction::ResourceBoundsMapping {
+    let mut map = BTreeMap::new();
+    map.insert(
+        starknet_api::transaction::Resource::L1Gas,
+        starknet_api::transaction::ResourceBounds { max_amount: 10000, max_price_per_unit: 12000 },
+    );
+    map.insert(
+        starknet_api::transaction::Resource::L1Gas,
+        starknet_api::transaction::ResourceBounds { max_amount: 50000, max_price_per_unit: 31000 },
+    );
+    starknet_api::transaction::ResourceBoundsMapping(map)
+}
+
+fn stark_felt_to_field_element(felt: &StarkFelt) -> FieldElement {
+    // Can only fail if the 32 byte number is not smaller than the field's modulus,
+    // but StarkFelt already enforces that
+    FieldElement::from_bytes_be(&felt.0).unwrap()
+}
+
+pub fn get_balance_contract_call(
+    sender_address: ContractAddress,
+    contract_address: ContractAddress,
+) -> Vec<Felt252Wrapper> {
+    let call_args = build_get_balance_contract_call(sender_address.0.0);
+    Starknet::call_contract(contract_address, call_args.0, call_args.1).unwrap()
+}
+
+pub fn set_sender_erc20_balance_to_zero(sender_address: ContractAddress, erc20_contract_address: ContractAddress) {
+    // Get balance variable key
+    let balance_low_storage_key = get_storage_key(
+        &erc20_contract_address,
+        "ERC20_balances",
+        &[stark_felt_to_field_element(&sender_address.0.0)],
+        0,
+    );
+
+    let balance_high_storage_key = get_storage_key(
+        &erc20_contract_address,
+        "ERC20_balances",
+        &[stark_felt_to_field_element(&sender_address.0.0)],
+        1,
+    );
+
+    // Set balance storage value to zero
+    StorageView::<MockRuntime>::insert(balance_low_storage_key, StarkFelt::try_from("0").unwrap());
+    StorageView::<MockRuntime>::insert(balance_high_storage_key, StarkFelt::try_from("0").unwrap());
+
+    // Ensure that balance is truely set to zero
+
+    pretty_assertions::assert_eq!(
+        get_balance_contract_call(sender_address, erc20_contract_address),
+        vec![Felt252Wrapper::from_hex_be("0x0").unwrap(), Felt252Wrapper::from_hex_be("0x0").unwrap()]
+    );
```

Comment:
There is no need to ensure at everycall. Just to write a unit-test for your method. If it works once, it will will work the others times